### PR TITLE
feat: add a progression view for media backup

### PR DESCRIPTION
**Pull request by enguerran**
We could use `Alerter` from [cozy-ui](https://github.com/cozy/cozy-ui) but it seems not correct for this kind of progression message (see @goldoraf and @gregorylegarec for more information).

So the `UploadProgression` component is the very same UI than an Alerter, but it did not have the same behavior, it is very simpler: if a message is display, it is shown, otherwise it is not.

Hope it fits the needs.

vgdmi9t

### CODE REVIEW

**Comment by enguerran** [comment]
I have to update the Alerter component for this feature.
But it is out of the scope.

The Alerter component's behavior is different in desktop view and in responsive view...

File: mobile/src/actions/mediaBackup.js [hunk 1]
```diff
@@ -8,10 +8,22 @@ import { HTTP_CODE_CONFLICT } from '../../../src/actions'
 export const MEDIA_UPLOAD_START = 'MEDIA_UPLOAD_START'
 export const MEDIA_UPLOAD_END = 'MEDIA_UPLOAD_END'
 export const IMAGE_UPLOAD_SUCCESS = 'IMAGE_UPLOAD_SUCCESS'
+export const CURRENT_UPLOAD = 'CURRENT_UPLOAD'
 
 export const startMediaUpload = () => ({ type: MEDIA_UPLOAD_START })
 export const endMediaUpload = () => ({ type: MEDIA_UPLOAD_END })
-export const successImageUpload = (image) => ({ type: IMAGE_UPLOAD_SUCCESS, id: image.id })
+export const successImageUpload = (media) => ({ type: IMAGE_UPLOAD_SUCCESS, id: media.id })
```
Comment (kosssi) [inline mobile/src/actions/mediaBackup.js]: why do you change (image) to (media)? (we have success**Image**Upload and **IMAGE**_UPLOAD_SUCCESS)

File: mobile/src/actions/mediaBackup.js [hunk 2]
```diff
@@ -33,8 +45,11 @@ export const mediaBackup = (dir) => async (dispatch, getState) => {
     let photos = await getFilteredPhotos()
     const alreadyUploaded = getState().mobile.mediaBackup.uploaded
     const dirID = await getDirID(dir)
+    const totalUpload = photos.length - alreadyUploaded.length
```
Comment (kosssi) [inline mobile/src/actions/mediaBackup.js]: It's not good here.
I think you can filter photos like that:

    photos = photos.filter(photo => !alreadyUploaded.includes(photo.id))
    const totalUpload = photos.length

and remove `if (!alreadyUploaded.includes(photo.id)) {` line 51

File: mobile/src/reducers/mediaBackup.js [hunk 3]
```diff
@@ -12,11 +12,13 @@ export const mediaBackup = (state = initialState, action) => {
     case MEDIA_UPLOAD_START:
       return { ...state, uploading: true }
     case MEDIA_UPLOAD_END:
-      return { ...state, uploading: false }
+      return { ...state, uploading: false, currentUpload: undefined }
     case IMAGE_UPLOAD_SUCCESS:
-      return { ...state, uploaded: [...state.uploaded, action.id] }
+      return { ...state, uploaded: [...state.uploaded, action.id], currentUpload: undefined }
```
Comment (kosssi) [inline mobile/src/reducers/mediaBackup.js]: if you do that (currentUpload: undefined) for each photo, the UploadProgression component will blinking...

Comment (enguerran) [inline mobile/src/reducers/mediaBackup.js]: Good remarks, it is an oversight.

File: mobile/src/actions/mediaBackup.js (hunk 2 above)
Comment (enguerran) [inline mobile/src/actions/mediaBackup.js]: I needed to get the total of *going to be uploaded* photos.

I think that 
```js
photos.length - alreadyUploaded.length === photos.filter(photo => !alreadyUploaded.includes(photo.id)).length
```

So this remark, though good enough, is just a matter of refactoring.

File: mobile/src/actions/mediaBackup.js (hunk 1 above)
Comment (enguerran) [inline mobile/src/actions/mediaBackup.js]: We don't backup images but photos. And soon we won't upload photos but photos and videos. So I think media is a good shell to address all the needs.

File: mobile/src/actions/mediaBackup.js (hunk 2 above)
Comment (kosssi) [inline mobile/src/actions/mediaBackup.js]: Precisely it is not the same.

If a user deletes already uploaded photos, they will count in `alreadyUploaded` but not in `photos`, which will distort the calculation.

Comment (enguerran) [inline mobile/src/actions/mediaBackup.js]: That is a very good remark. Please find https://github.com/cozy/cozy-files-v3/pull/184/commits/77a6a491cd71403a6125573ea5b5d93669a7e69b to fix this.

**Comment by enguerran** [comment]
@kosssi di you have something else to add or do you approve this PR?

File: mobile/src/actions/mediaBackup.js [hunk 4]
```diff
@@ -27,21 +39,25 @@ async function getDirID (dir) {
 
 export const mediaBackup = (dir) => async (dispatch, getState) => {
   if (backupAllowed(getState().mobile.settings.wifiOnly)) {
-    let photos = await getFilteredPhotos()
+    const photosOnDevice = await getFilteredPhotos()
```
Comment (kosssi) [inline mobile/src/actions/mediaBackup.js]: 👍

**Comment by enguerran** [comment]
Alright

**Pull request merged**